File: api/src/test/java/net/blueshell/api/factory/dto/FileDTOFactory.java
package net.blueshell.api.factory.dto;

import lombok.RequiredArgsConstructor;
import net.blueshell.api.common.enums.FileType;
import net.blueshell.api.dto.FileDTO;
import org.springframework.stereotype.Component;

/**
 * Factory for FileDTO test instances.
 */
@Component
@RequiredArgsConstructor
public class FileDTOFactory extends BaseDtoFactory<FileDTO> {

    @Override
    public Class<FileDTO> targetType() {
        return FileDTO.class;
    }

    @Override
    public FileDTO createBasic() {
        FileDTO dto = new FileDTO();
        dto.setName(unique("file"));
        dto.setMediaType("image/jpeg");
        dto.setSize(1024L);
        dto.setType(FileType.EVENT_BANNER);
        return dto;
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/dto/FileDTOFactory.kt
package net.blueshell.api.factory.dto

import net.blueshell.api.domain.file.web.dto.FileDTO
import net.blueshell.api.shared.enums.FileType
import org.springframework.stereotype.Component

/**
 * Factory for FileDTO test instances.
 */
@Component
class FileDTOFactory : BaseDtoFactory<FileDTO>() {

    override fun targetType(): Class<FileDTO> = FileDTO::class.java

    override fun createBasic(): FileDTO {
        return FileDTO(
            name = unique("file"),
            mediaType = "image/jpeg",
            size = 1024L,
            type = FileType.EVENT_BANNER
        )
    }
}
